pkg/controllermanager/controller/project/project.go

```diff
@@ -149,23 +132,28 @@ func NewProjectController(
 	})
 
 	shootInformer.AddEventHandler(cache.ResourceEventHandlerFuncs{
-		AddFunc: projectController.updateShootActivity,
+		AddFunc:    func(obj interface{}) { projectController.updateShootActivity(ctx, obj) },
```

Comment:
Afaik there is an `ADD` event on every controller start. Thus, in combination with the Reconciler which just updates the `LastActivityTimestamp` on each run, it would be refreshed on every start.

Reply:
Right, isn't this what we want? :)

Reply:
Should it be updated when the controller crashes or rolls?

Reply:
The correct solution would be using the maximum of `creationTimestamp` and `lastActivityTimestamp` and `lastActivityTimestamp` on ADD events and the the current time on UPDATE events. However, there is no straight forward solution to bring the ADD event information the `Reconcile` method.

How about this simplified approach
- Trigger a reconcile on UPDATE event when `spec` is changing
- Trigger a reconcile on ADD event when object is newly created. Maybe `creationTimestamp > "(now - 1h)"`
- Set `lastActivityTimestamp` to "now" on every reconcile (as proposed)

With that we could minimise the consequences of setting false `lastActivityTimestamp` on ADD events. We can  be wrong by 1h maximum .
Additionally, we don't lose ADD events when the controller is down for a short period.
`1h` might not be the optimal value. It is an assumption. I don't know for how long the project activity controller could be down. Maybe `1d` would be still good enough for identification of stale projects.

Reply:
Thanks @oliver-goetz for proposing alternatives, very much appreciated! I don't have any better proposal right now, so maybe we can indeed start with this (probably `1h` is good enough for now) and refine this as needed in the future?

Reply:
A thing i forgot. 
DELETE events should trigger a reconcile too. Otherwise, projects which haven't been touched for a while could get stale immediately when the last object was deleted

Reply:
Not sure, I think it makes sense to add the DELETE handlers (this isn't too much work/additional code anyways) to prevent surprising behaviour and `Project`s which are marked as "stale" too fast/early. :)